From 7d9df199e9ec743540ce42cc967530decfa17836 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Oskari=20Lemmel=C3=A4?= <oskari@lemmela.net>
Date: Sun, 26 Jul 2026 21:26:44 +0300
Subject: [PATCH 1/2] dt-bindings: net: pse-pd: add MikroTik PoE controller
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Describe the SPI-connected MikroTik RouterBOARD PoE controller on
RB960PGS boards only. Require SPI mode 0 and expose four PSE power
interfaces, numbered 0 through 3, for Ethernet PHY pses references.

Signed-off-by: Oskari Lemmelä <oskari@lemmela.net>
---
 .../net/pse-pd/mikrotik,routerboard-poe.yaml  | 93 +++++++++++++++++++
 1 file changed, 93 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/pse-pd/mikrotik,routerboard-poe.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/net/pse-pd/mikrotik,routerboard-poe.yaml
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/pse-pd/mikrotik,routerboard-poe.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: MikroTik RouterBOARD PoE PSE controller
+
+maintainers:
+  - Oskari Lemmelä <oskari@lemmela.net>
+
+description:
+  MikroTik RouterBOARD PoE Power Sourcing Equipment controller on
+  RB960PGS boards. The controller communicates over SPI using a
+  CRC8-protected command protocol and controls four PoE output ports.
+  SPI mode 0 is required. Passive PoE and 802.3af/at output are supported
+  depending on the supply voltage.
+
+allOf:
+  - $ref: pse-controller.yaml#
+  - $ref: /schemas/spi/spi-peripheral-props.yaml#
+
+properties:
+  compatible:
+    enum:
+      - mikrotik,rb960pgs
+
+  reg:
+    maxItems: 1
+
+  spi-cpha: false
+  spi-cpol: false
+
+  pse-pis:
+    type: object
+    description:
+      PSE Power Interfaces, one per PoE port. Ports are numbered 0 to 3
+      and must be referenced by the corresponding ethernet-phy nodes using
+      the 'pses' property.
+
+    patternProperties:
+      "^pse-pi@[0-9a-f]+$":
+        type: object
+        properties:
+          reg:
+            items:
+              minimum: 0
+              maximum: 3
+
+required:
+  - compatible
+  - reg
+  - pse-pis
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    spi {
+      #address-cells = <1>;
+      #size-cells = <0>;
+
+      ethernet-pse@0 {
+        compatible = "mikrotik,rb960pgs";
+        reg = <0>;
+        spi-max-frequency = <2100000>;
+
+        pse-pis {
+          #address-cells = <1>;
+          #size-cells = <0>;
+
+          pse_pi0: pse-pi@0 {
+            reg = <0>;
+            #pse-cells = <0>;
+          };
+
+          pse_pi1: pse-pi@1 {
+            reg = <1>;
+            #pse-cells = <0>;
+          };
+
+          pse_pi2: pse-pi@2 {
+            reg = <2>;
+            #pse-cells = <0>;
+          };
+
+          pse_pi3: pse-pi@3 {
+            reg = <3>;
+            #pse-cells = <0>;
+          };
+        };
+      };
+    };
